azalea-chat/src/component.rs

```diff
@@ -89,41 +135,26 @@ impl FormattedText {
     /// println!("{}", component.to_ansi());
     /// ```
     pub fn to_ansi(&self) -> String {
-        // default the default_style to white if it's not set
-        self.to_ansi_with_custom_style(&DEFAULT_STYLE)
+        self.to_custom_format(
+            |running, new, default| (running.compare_ansi(new, default), "".to_owned()),
+            |text| text.to_string(),
+            |style| if !style.is_empty() { "\u{1b}[m" } else { "" }.to_string(),
+            &DEFAULT_STYLE,
+        )
     }
 
-    /// Convert this component into an
-    /// [ANSI string](https://en.wikipedia.org/wiki/ANSI_escape_code).
-    ///
-    /// This is the same as [`FormattedText::to_ansi`], but you can specify a
-    /// default [`Style`] to use.
-    pub fn to_ansi_with_custom_style(&self, default_style: &Style) -> String {
```

Comment:
I'd like to keep `to_ansi_with_custom_style` to avoid breaking changes and so if the user wants a custom default style then they don't have to copy the code from `to_ansi`.

Btw, the reason this exists at all is to make it easier if the user is doing something like logging all of their text in a certain color and they want to make sure the text stays that color when `reset` gets set to true. This doesn't matter for HTML since in HTML you can just make a parent element with the default style.